corehq/apps/cloudcare/views.py

```diff
@@ -142,7 +145,12 @@ def set_cookie(response):  # set_coookie is a noop by default
             'restoreAs:{}:{}'.format(domain, request.couch_user.username))
         username = request.COOKIES.get(cookie_name)
         if username:
-            user = CouchUser.get_by_username(format_username(username, domain))
+            username = urllib.parse.unquote(username)
+            # mobile workers do not have special characters in their name
```

Comment:
I think this logic should go in `corehq.apps.users.util`. It's basically the same as `get_complete_mobile_username`, although it has a different purpose, so if you were to reuse that function it ought to be named something else - `get_username_for_restore_as`?

Reply:
@orangejenny Will use `get_complete_mobile_username`. Going by the comment on it I feel pretty good about using the @. Since the usernames should already be validated at this point. Do you think I should move the url decoding in the utils file? I am a bit torn. It seems like the responsibility of the view.